### PR TITLE
ComboBox: fix prop usage

### PR DESCRIPTION
**Pull request by jspurlin**
#### Pull request checklist

- [X] Include a change request file using `$ npm run change`

#### Description of changes
With the introduction of onChange, the ComboBox's wrapping div will also get the onChange if supplied resulting in potentially much more frequent (and unneeded/unexpected) calls to onChange. I also noticed that if value is supplied it is also applied to the wrapping div when it shouldn't.

When debugging the onChange issue I also realized that a ComboBox could potentially attempt to submit a pending value after it had already submitted the pending value and called clear pending value (for example, the TAB followed by blur case). Logically clearing the pending value will make it so the second submit pending value no-ops since it calls setState to clear the pending values, but setState is not immediate and we can get into a case where submitPending value (and hence onChange) is called twice.

There is a callback for setState which runs after the state has actually been set. I've added processingClearPendingInfo to denote when we've called clearPendingInfo but the state has yet to be set. This makes sure that if we get back into submitPendingValue while we have already called clearPendingValue, we don't need to do anything. The could be hit more easily for ComboBoxes inside of FocusZones.

#### Focus areas to test
Verified that we get the onChange as expected when the value changes both for freeform values (when allowFreeform is true) as well as selecting an existing option. Everything also looks good regardless of if the comboBox is inside a FocusZone or not. Also validated the different potential ways for submitting a value, TAB,ENTER

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx [hunk 1]
```diff
@@ -517,11 +520,17 @@ export class ComboBox extends BaseComponent<IComboBoxProps, IComboBoxState> {
 
     // If the user passed is a value prop, use that
     // unless we are open and have a valid current pending index
-    if (!(isOpen && currentPendingIndexValid) && (text && !currentPendingValue)) {
+    if (
+      !(isOpen && currentPendingIndexValid) &&
+      (text && (currentPendingValue === null || currentPendingValue === undefined))
```
Comment (chang47) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: >(currentPendingValue === null || currentPendingValue === undefine [](start = 15, length = 65)

Why was this changed? Are we accounting for the case where the currentPendingValue is "" which translates to false?

Comment (jspurlin) [reply packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx]: Exactly so if a user is typing a freeform value and they get to an empty string it could (if value or text are present) revert to that value when it should still show an empty string
